src/sentry/api/endpoints/project_environments.py

```diff
@@ -4,22 +4,21 @@
 
 from sentry.api.bases.project import ProjectEndpoint
 from sentry.api.serializers import serialize
-from sentry.models import Environment
+from sentry.models import EnvironmentProject
 
 
 environment_visibility_filter_options = {
     'all': lambda queryset: queryset,
-    'hidden': lambda queryset: queryset.filter(environmentproject__is_hidden=True),
-    'visible': lambda queryset: queryset.exclude(environmentproject__is_hidden=True),
+    'hidden': lambda queryset: queryset.filter(is_hidden=True),
+    'visible': lambda queryset: queryset.exclude(is_hidden=True),
```

Comment:
Is it more straightforward to say `filter(is_hidden=False)` or is `is_hidden` nullable as well?

If they both used the same method on the queryset, you might not need these lambdas either.

Reply:
`is_hidden` is nullable since it was added in GH-7087 after data already existed in the table. 😞  I agree this `filter`/`exclude` dichotomy is kind of weird.